### PR TITLE
FG-3744: Support User-Provided Secrets in Helm Charts

### PR DESCRIPTION
**Pull request by foxglovenes**
### User-Facing Changes

Customers deploying a primary site or an edge site will be able to define Kubernetes Secrets to load as environment variables by listing them in `.Values.globals.secrets`. This is primarily so that customers do not have to include the Foxglove site token in plain text in values.yaml, however it allows for supporting other customer-provided secrets. The existing method of using `.Values.globals.site_token` will still work until we can safely remove it.

### Description

Adds a new section in values.yaml to define Kubernetes Secrets that will be loaded into the environment for all containers via   envFrom directives. Wraps all of the code that currently depends on `.Values.globals.site_token` in conditional blocks so that if `site_token` is blank they are not evaluated. The edge controller now supports both FOXGLOVE_SITE_TOKEN and SITE_TOKEN.

### CODE REVIEW

**Comment by foxglovenes** [comment]
@defunctzombie Updated the PR description.

**Review by defunctzombie — APPROVED** [review]
LGTM - you'll need to bump the chart version for edge and primary in this PR otherwise the github action will fail to make a new release for the changes

**Comment by foxglovenes** [comment]
Yes, I was waiting on an approval before that happened so I wouldn't have to bump it twice.

**Pull request merged**
